fastfield_codecs/src/multilinearinterpol_v2.rs

```diff
@@ -0,0 +1,376 @@
+/*!
+
+MultiLinearInterpolV2 compressor uses blockwise linear interpolation to guess values and stores the difference between the actual value
+and the one given by the linear interpolation.
+This is done for every block of 512 values. For every block, our linear function can be expressed as
+`computed_value = slope * block_position + first_value + positive_offset`
+where:
+- `block_position` is the position inside of the block from 0 to 511
+- `first_value` is the first value on the block
+- `positive_offset` is computed such that we ensure the diff `real_value - computed_value` is always positive.
+
+21 bytes is needed to store the block metadata, it adds an overhead of 21 * 8 / 512 = 0,33 bits per element.
+
+*/
+
+use crate::FastFieldCodecReader;
+use crate::FastFieldCodecSerializer;
+use crate::FastFieldDataAccess;
+use crate::FastFieldStats;
+use std::io::{self, Read, Write};
+use std::ops::Sub;
+use tantivy_bitpacker::compute_num_bits;
+use tantivy_bitpacker::BitPacker;
+
+use common::BinarySerializable;
+use common::DeserializeFrom;
+use tantivy_bitpacker::BitUnpacker;
+
+const BLOCK_SIZE: u64 = 512;
+
+#[derive(Clone)]
+pub struct MultiLinearInterpolV2FastFieldReader {
```

Comment:
I suggest we stop calling this MultiLinear. MultiLinear usually means linear against with regard to several variables.
PiecewiseLinear is the correct terminology I believe.